src/main/java/com/shaft/tools/io/PdfFileManager.java

```diff
@@ -0,0 +1,173 @@
+package com.shaft.tools.io;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.pdfbox.cos.COSDocument;
+import org.apache.pdfbox.io.RandomAccessBufferedFileInputStream;
+import org.apache.pdfbox.pdfparser.PDFParser;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.text.PDFTextStripper;
+import org.testng.Assert;
+
+public class PdfFileManager {
+
+	private RandomAccessBufferedFileInputStream stream = null;
+	private PDFParser parser = null;
+	private COSDocument cosDoc = null;
+	private PDFTextStripper strip = null;
+	private File file;
+
+	/**
+	 * Validates the downloaded pdf document from the site under test
+	 * 
+	 * @param url                the path of the downloaded pdf file (usually will
+	 *                           be target/downloadedFiles/your-file.pdf), usually
+	 *                           will be in "target/downloadedFiles/" for the
+	 *                           project
+	 * @param milliSecondsToWait the time in milliseconds needed to wait on the
+	 *                           document to be downloaded successfully to the
+	 *                           target folder path
+	 */
+	public PdfFileManager(String url, int milliSecondsToWait) {
+		// Waiting for file to download
+		waitOnDocumentToBeReady(milliSecondsToWait);
+
+		file = new File(url);
+	}
+
+	public enum DeleteFileAfterValidationStatus {
+		TRUE, FALSE
+	}
+
+	/**
+	 * @param startPageNumber                 the starting page for the document to
+	 *                                        be validated
+	 * @param endPageNumber                   the ending page for the document to be
+	 *                                        validated
+	 * @param deleteFileAfterValidationStatus the status of deleting the file after
+	 *                                        get the document text
+	 * @return returns the pdf content in string so that can be validated
+	 */
+	public String readPDFContentFromDownloadedPDF(int startPageNumber, int endPageNumber,
+			DeleteFileAfterValidationStatus deleteFileAfterValidationStatus) {
+
+		stream = readFileInputStream(file);
+		parser = parseStreamDocument(stream);
+
+		cosDoc = getParsedDocument(parser);
+		String content = getPdfText(cosDoc, startPageNumber, endPageNumber);
+		closeStreamAndDeleteFile(file, stream, deleteFileAfterValidationStatus);
+
+		return content;
+	}
+
+	private void waitOnDocumentToBeReady(int milliSeconds) {
```

Comment:
Better than thread.sleeping for the entire duration, there are methods in the FileActions library that checks if a file exists.
You can have a polling interval, and go check every 5 seconds until the file exists or the timeout is reached.
This is how the selenium fluent wait works, but we'll implement it from scratch.
Better to check and see if there's already a method in SHAFT that waits until a file exists already.

And as agreed, timeouts will be provided from the property file and not as a method parameter

Reply:
@MohabMohie 
can we do it like this in FileActions?

if (Boolean.FALSE.equals(doesFileExit)) {
				try {
					Thread.sleep(TimeUnit.SECONDS.toMillis(Long.parseLong(System.getProperty("fileTimeout"))));
				} catch (Exception e1) {
					ReportManagerHelper.log(e1);
				}
			}

Reply:
this is still one thread.sleep for the entire timeout duration. but we need to have it check every 5 seconds until it finds the file or the timeout is reached.